[BUG] : unable to set field_delimiter_regex

**Describe the bug**
While trying to set field_delimiter_regex, I see the following error `Caused by: java.lang.IllegalArgumentException: field_delimiter_regex and field_split_characters cannot both be defined.`, I am not setting field_split_characters, I assume it is using the default one.

**To Reproduce**

1. Use the configuration mentioned in the  [documentation](https://opensearch.org/docs/latest/data-prepper/pipelines/configuration/processors/key-value/)
`field_delimiter_regex: "&\\{2\\}"`
2. Star the docker container
```
docker run --rm --name data-prepper -p 2021:2021-v $(pwd)/pipeline.yaml:/usr/share/data-prepper/pipelines/pipelines.yaml -v $(pwd)/data-prepper-config.yaml:/usr/share/data-prepper/config/data-prepper-config.yaml -v data-prepper-out:/usr/share/log opensearchproject/data-prepper:latest
```
4. See error
```
Caused by: java.lang.IllegalArgumentException: field_delimiter_regex and field_split_characters cannot both be defined.
	at org.opensearch.dataprepper.plugins.processor.keyvalue.KeyValueProcessor.<init>(KeyValueProcessor.java:45) ~[key-value-processor-2.2.1.jar:?]
	at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77) ~[?:?]
	at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:?]
	at java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499) ~[?:?]
	at java.lang.reflect.Constructor.newInstance(Constructor.java:480) ~[?:?]
	at org.opensearch.dataprepper.plugin.PluginCreator.newPluginInstance(PluginCreator.java:40) ~[data-prepper-core-2.2.1.jar:?]
	... 35 more
```

**Pipeline config**
```
log-pipeline:
  source:
    http:
  processor:
    - key_value:
        field_delimiter_regex: "&\\{2\\}"
        source: "log"
  sink:
    - file:
        path: /usr/share/log/output-file
```

--- a/data-prepper-plugins/key-value-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/keyvalue/KeyValueProcessorConfig.java
+++ b/data-prepper-plugins/key-value-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/keyvalue/KeyValueProcessorConfig.java
@@ -40,7 +40,6 @@ public class KeyValueProcessorConfig {
     private String fieldDelimiterRegex;
 
     @JsonProperty("field_split_characters")
-    @NotEmpty
     private String fieldSplitCharacters = DEFAULT_FIELD_SPLIT_CHARACTERS;
 
     @JsonProperty("include_keys")

